adafruit_mcp3xxx/analog_in.py

```diff
@@ -54,15 +54,14 @@ def __init__(
 
     @property
     def value(self) -> int:
-        """Returns the value of an ADC pin as an integer. Due to 10-bit accuracy of the chip, the
-        returned values range [0, 65472]."""
-        return (
-            self._mcp.read(self._pin_setting, is_differential=self.is_differential) << 6
+        """Returns the value of an ADC pin as an integer in the range [0, 65535]."""
+        return int(
+            self._mcp.read(self._pin_setting, is_differential=self.is_differential)
+            * (65535 / 1023)
```

Comment:
This incorporates @tannewt's suggestion to use a slightly different technique:
```suggestion
        # Initial result is only 10 bits.
        result = int(self._mcp.read(self._pin_setting, is_differential=self.is_differential))
        # Stretch to 16 bits and cover full range.
        return (result << 6) | (result >> 4)
```